src/notation/view/percussionpanel/percussionpanelmodel.cpp

```diff
@@ -186,16 +186,11 @@ void PercussionPanelModel::finishEditing(bool discardChanges)
     Drumset updatedDrumset = *m_padListModel->drumset();
     m_padListModel->removeEmptyRows();
 
-    NoteInputState inputState = interaction()->noteInput()->state();
-    const Staff* staff = inputState.staff;
-
-    IF_ASSERT_FAILED(staff && staff->part()) {
-        return;
-    }
-
-    Instrument* inst = staff->part()->instrument(inputState.segment->tick());
+    const std::pair<Instrument*, Part*> instAndPart = getCurrentInstrumentAndPart();
+    Instrument* inst = instAndPart.first;
+    Part* part = instAndPart.second;
```

Comment:
If you want, you can do
```suggestion
    const auto [inst, part] = getCurrentInstrumentAndPart();
```
(if I'm not mistaken)

Reply:
I think in this case I'd prefer to leave it as-is (just so the types of `inst` and `part` are super obvious)